src/main/java/org/apache/maven/plugin/compiler/ModuleInfoTransformer.java

```diff
@@ -45,10 +53,13 @@ public ModuleVisitor visitModule(String name, int access, String version) {
                     @Override
                     public void visitRequire(String module, int access, String version) {
                         // Check if the module name matches the java/jdk modules
-                        if (module.startsWith("java.") || module.startsWith("jdk.")) {
+                        if (version != null
+                                && !version.equals(javaVersion)
```

Comment:
Tried that already, but it doesn't work, the initial version of every major JDK release uses a simple version (17) and doesn't have a patch version (eg. 17.0.0), and this is problematic due to the constant pool reorder making it unreproducible on the first release.

Reply:
nice catch, I did not expect `[INFO] JDK-8318913 workaround: patched module-info.class requires version from [21] to [21]` to be useful, but yes, tested with an EA version, and also found why checking checksum was also useful
thank you